site/cypress/e2e/Cypress-Autograding/electronic_gradeable.spec.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Uploads a file and compares the results with expected scores.
+ * @param {string} fileUploadName the fixture to upload
+ * @param {number[]} fullScores the max scores for the gradeables
+ * @param {number[]} expectedScores the expected score for the submission
+ */
+const submitAndCheckResults = (fileUploadName, expectedScores, fullScores) => {
+    expect(expectedScores.length).to.eq(fullScores.length);
+    const scoreTotal = fullScores.reduce((partial, actual) => partial + actual, 0);
+    const expectedTotal = expectedScores.reduce((partial, actual) => partial + actual, 0);
+
+    // Wait for client JS to load - reduces flakyness
+    cy.get('[data-testid="gradeable-time-remaining-text"]').contains('days');
+    // If the clear button exists, we should click it.
```

Comment:
What about gradeables that have multiple "buckets" - e.g., cpp_simple_lab, choice_of_language, and others